src/components/web-components/rank-card/rank-card.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { View, Text } from 'react-native';
+import stylesProvider from './styles';
+import { IThemeProps, withTheme } from '../../../core/theme/with-theme';
+
+export const RankCardComponent = (props: IThemeProps<ReturnType<typeof stylesProvider>>) => {
```

Comment:
I thins component, you should send the hardcoded values as ExternalProps.

Also you can make this component a little smarter and so that you don't duplicate a lot of code.
This is only an example, you choose what labels and values helps you. 

```
export interface IExternalProps {
    title1: string;
    value1: string;

    title2: string;
    value2: string;

    ...
}

export const RankCardComponent = (
    props: IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>
) => {
    const { styles } = props;

    const renderRow = (title: string, subtitle: string) => {
        return (
            <View style={styles.column}>
                <Text style={styles.title}>{title}</Text>
                <Text style={styles.value}>{subtitle}</Text>
            </View>
        );
    };

    return (
        <View style={styles.container}>
            {renderRow(props.title1, props.value1)}
            {renderRow(props.title2, props.value2)}

            ...
        </View>
    );
};

export const RankCard = smartConnect<IExternalProps>(RankCardComponent, [
    withTheme(stylesProvider)
]);
```